src/sat/smt/user_solver.cpp

```diff
@@ -43,15 +43,16 @@ namespace user_solver {
             m_prop.push_back(prop_info(explain, v, r));
     }
 
-    void solver::propagate_cb(
-        unsigned num_fixed, expr* const* fixed_ids,
-        unsigned num_eqs, expr* const* eq_lhs, expr* const* eq_rhs,
-        expr* conseq) {
+    bool solver::propagate_cb(
+            unsigned num_fixed, expr* const* fixed_ids,
+            unsigned num_eqs, expr* const* eq_lhs, expr* const* eq_rhs,
+            expr* conseq) {
         m_fixed_ids.reset();
         for (unsigned i = 0; i < num_fixed; ++i)
```

Comment:
Seems you want the following;

auto* n = ctx.get_enode(conseq);
if (n && s().value(enode2literal(n)) == l_true)
   return false;